dsc/src/util.rs

```diff
@@ -268,46 +287,78 @@ pub fn write_output(json: &str, format: &Option<OutputFormat>) {
     }
 }
 
-pub fn enable_tracing(trace_level: &Option<TraceLevel>, trace_format: &TraceFormat) {
-    let tracing_level = match trace_level {
-        Some(level) => level,
-        None => {
-            // use DSC_TRACE_LEVEL env var if set
-            match env::var(DSC_TRACE_LEVEL) {
-                Ok(level) => {
-                    match level.to_ascii_uppercase().as_str() {
-                        "ERROR" => &TraceLevel::Error,
-                        "WARN" => &TraceLevel::Warn,
-                        "INFO" => &TraceLevel::Info,
-                        "DEBUG" => &TraceLevel::Debug,
-                        "TRACE" => &TraceLevel::Trace,
-                        _ => {
-                            warn!("Invalid DSC_TRACE_LEVEL value '{level}', defaulting to 'warn'");
-                            &TraceLevel::Warn
-                        },
-                    }
+pub fn enable_tracing(trace_level_arg: Option<TraceLevel>, trace_format_arg: Option<TraceFormat>) {
```

Comment:
Was there a specific reason to not pass by reference?  Also, I don't think we should have the `_arg` suffix as it starts looking more like hungarian

Reply:
"Was there a specific reason to not pass by reference?"
negligible perf impact here - these are options of tiny enums, not much benefit of using references.

"I don't think we should have the _arg suffix as it starts looking more like hungarian"
The _arg suffix is supposed to mean "value from dsc command-line argument".
I'm open to other suggestions.

Reply:
Agree the perf difference may be negligible, but future developers may be confused as we don't have a consistent pattern.  I think we should default to passing by ref unless we have a good reason not to.

For the param name, I think you should just drop the `_arg` suffix and use variable shadowing (same name) inside the function.

Reply:
Updated.